compat/krb5.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef _SQUID__COMPAT_KRB5_H_
+#define _SQUID__COMPAT_KRB5_H_
+
+/*
+ * The Kerberos library krb5.h header file has various major
+ * bugs in some implementations:
+ *
+ * - MacOS marks as deprecated the POSIX compatible APIs.
+ *   Apparently to encourage code to use their internal APIs.
+ *
+ * - Heimdal may lack the extern "C" syntax for building in C++
+ *
+ * - Solaris incorrectly implements its own the extern "C" replacment
+ *   macros and as a result krb5.h contains a trailing '}'.
+ *   see http://bugs.opensolaris.org/bugdatabase/view_bug.do?bug_id=6837512
+ *
+ * This file exists to fix those issues as best we can and to
+ * ensure the logic is identical in Squid code, Squid helpers,
+ * and autoconf tests. see acinclude/krb5.h for the later.
```

Comment:
Grammar(?) and spelling nitpicks:

```suggestion
 * and autoconf tests. See acinclude/krb5.h for the latter.
```